src/main/java/mat/client/shared/MeasureLibraryGridToolbar.java

```diff
@@ -0,0 +1,163 @@
+
+import java.util.Collection;
+
+import org.gwtbootstrap3.client.ui.Button;
+import org.gwtbootstrap3.client.ui.constants.ButtonSize;
+import org.gwtbootstrap3.client.ui.constants.ButtonType;
+import org.gwtbootstrap3.client.ui.constants.IconType;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.thirdparty.guava.common.annotations.VisibleForTesting;
+import mat.client.measure.ManageMeasureSearchModel;
+
+public class MeasureLibraryGridToolbar extends HorizontalFlowPanel {
+
+    private Button versionButton;
+    private Button historyButton;
+    private Button editButton;
+    private Button shareButton;
+    private Button cloneButton;
+    private Button exportButton;
+
+    public MeasureLibraryGridToolbar() {
+        setStyleName("action-button-bar");
+        addStyleName("btn-group");
+        addStyleName("btn-group-sm");
+
+        versionButton = GWT.create(Button.class);
+        historyButton = GWT.create(Button.class);
+        editButton = GWT.create(Button.class);
+        shareButton = GWT.create(Button.class);
+        cloneButton = GWT.create(Button.class);
+        exportButton = GWT.create(Button.class);
+
+        add(versionButton);
+        add(historyButton);
+        add(editButton);
+        add(shareButton);
+        add(cloneButton);
+        add(exportButton);
+
+
+        applyDefault();
+    }
+
+    @VisibleForTesting
+    void applyDefault() {
+        applyDefaultAllButExport();
+
+        exportButton.setText("Export");
+        exportButton.setTitle("Click to export");
+        exportButton.setWidth("72px");
+        buildActionButton(exportButton, IconType.DOWNLOAD);
+    }
+
+    private void applyDefaultAllButExport() {
+        versionButton.setText("Create Version/Draft");
+        versionButton.setTitle("Click to create version or draft");
+        versionButton.setWidth("146px");
+        buildActionButton(versionButton, IconType.STAR);
+
+        historyButton.setText("History");
+        historyButton.setTitle("Click to view history");
+        historyButton.setWidth("73px");
+        buildActionButton(historyButton, IconType.CLOCK_O);
+
+        editButton.setText("Edit");
+        editButton.setTitle("Click to edit");
+        editButton.setWidth("57px");
+        buildActionButton(editButton, IconType.EDIT);
+
+        shareButton.setText("Share");
+        shareButton.setTitle("Click to share");
+        shareButton.setWidth("68px");
+        buildActionButton(shareButton, IconType.SHARE_SQUARE);
+
+        cloneButton.setText("Clone");
+        cloneButton.setTitle("Click to clone");
+        cloneButton.setWidth("69px");
+        buildActionButton(cloneButton, IconType.CLONE);
+    }
+
+    private Button buildActionButton(Button actionButton, IconType icon) {
+        actionButton.setType(ButtonType.DEFAULT);
+        actionButton.setSize(ButtonSize.SMALL);
+        actionButton.setIcon(icon);
+        actionButton.setEnabled(false);
+        return actionButton;
```

Comment:
doesn't use returned value anywhere. Can this have void return type?

Reply:
yes, you're right